service/history/historyEngine.go

```diff
@@ -2317,7 +2317,9 @@ func (e *historyEngineImpl) ResetWorkflowExecution(
 		request.GetWorkflowTaskFinishEventId() >= baseMutableState.GetNextEventID() {
 		return nil, serviceerror.NewInvalidArgument("Workflow task finish ID must be > 1 && <= workflow next event ID.")
 	}
-
+	if baseRunID == "" {
+		baseRunID = baseMutableState.GetExecutionInfo().RunID
+	}
```

Comment:
Why do we need this when the code that follows already loads the runID?

Reply:
basically we are saying that on like 2334, we can do this, right?:

if baseRunID == "" || currentRunID == baseRunID  {
}

Reply:
is this suggesting moving this if statement to after like 2330 and assigning it to currentRunID if baseRunID == ""?

Reply:
I think Max's point is that we already have a variable called currentRunID below and we could use it if baseRunID is empty.
Let me change this code a bit and see if it would look better at the end.

Reply:
I rewrote it in a way that we use currentRunID.